Userland/Libraries/LibWeb/CSS/CSSStyleDeclaration.cpp

```diff
@@ -256,14 +266,22 @@ String CSSStyleDeclaration::get_property_value(StringView property_name) const
             // 3. Append the declaration to list.
             if (!list.is_empty())
                 list.append(' ');
+            if (all_values_are_equal && last_serialized_value && *last_serialized_value != *declaration->value)
+                all_values_are_equal = false;
             list.append(declaration->value->to_string());
+            last_serialized_value = declaration->value;
 
             if (last_important_flag.has_value() && declaration->important != *last_important_flag)
                 return {};
             last_important_flag = declaration->important;
         }
 
         // 3. If important flags of all declarations in list are same, then return the serialization of list.
+        // https://drafts.csswg.org/cssom/#serialize-a-css-value
+        // If this algorithm is invoked with a list list:
+        //   Let shorthand be the first shorthand property, in preferred order, that exactly maps to all of the longhand properties in list.
```

Comment:
I see why you added this comment, but it's not describing what you're doing, and it's also not the full algorithm, so please remove it. We'd want a separate `serialize_a_css_value()` function (probably in `LibWeb/CSS/Serialize.h`) which follows that spec properly, and is called here. (Or at least, does something roughly equivalent - that algorithm looks very awkward (if not impossible) to implement directly.)